core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -249,9 +249,17 @@ public Builder setCurrentVersionId(int newVersionId) {
         changes.add(new MetadataUpdate.SetCurrentViewVersion(newVersionId));
       }
 
+      // Use the timestamp from the view version if it was added in current set of changes.
+      // Otherwise, use the current system time. This handles cases where the view version
+      // was set as current in the past and is being re-activated.
+      boolean versionAddedInThisChange =
+          changes(MetadataUpdate.AddViewVersion.class)
+              .anyMatch(added -> added.viewVersion().versionId() == newVersionId);
+
       this.historyEntry =
           ImmutableViewHistoryEntry.builder()
-              .timestampMillis(version.timestampMillis())
+              .timestampMillis(
```

Comment:
in hindsight I'm actually thinking whether it's more correct to always set this to `System.currentTimeMillis()` after re-reading the spec in https://iceberg.apache.org/view-spec/#version-log. @rdblue wdyt?

Reply:
If the version was added in the current request to change current-version-id, then it should use the timestamp from the version so it all matches. With timestamps there is no "correct" -- we want to have predictable and reasonable behavior. Making the timestamps match makes sense so that you don't have to reason about things that were "probably" part of the same change.

Reply:
thanks for the input here @rdblue